### ISSUE TITLE
RPMTAG_FILECLASS type mismatch

### ISSUE THREAD

**Issue by dmach** [rpm-software-management/rpm#1563]
The following assignment doesn't work due to type incompatibility:
```
new_hdr[rpm.RPMTAG_FILECLASS] = old_hdr[rpm.RPMTAG_FILECLASS]
```

rpmtag.h claims that it is a list of integers:
```
RPMTAG_FILECLASS        = 1141, /* i[] */
```

but the content are strings:
$ rpm -q bash --qf '[%{fileclass}\n]'
```
troff or preprocessor input, ASCII text (gzip compressed data, max compression, from Unix)
troff or preprocessor input, ASCII text (gzip compressed data, max compression, from Unix)
troff or preprocessor input, ASCII text (gzip compressed data, max compression, from Unix)
troff or preprocessor input, ASCII text (gzip compressed data, max compression, from Unix)
troff or preprocessor input, ASCII text (gzip compressed data, max compression, from Unix)
symbolic link to `bashbug.1.gz'
troff or preprocessor input, ASCII text (gzip compressed data, max compression, from Unix)
...
```

**Comment by pmatilai** [comment]
Yup, this unfortunate collision is actually even documented now :sweat_smile:  
https://github.com/rpm-software-management/rpm/blob/master/doc/manual/tags.md

There's no good way to fix the collision between the fundamental tag and the extension, because one way or the other, it'd break one thing or the other. However it's annoying that there's no way to access the raw data from either python or command line (except for `--xml` query) at the moment.

**Comment by ffesti** [comment]
I wonder if we could overload the assignment in the Python binding to translate the list of strings back into the dict it originally were.

**Comment by pmatilai** [comment]
Then we'd have to do the same for any number of other tags as well...
Header manipulation from Python is little more than a toy ATM.

**Comment by DemiMarie** [comment]
Can we at least fix this in RPMv6?

**Comment by pmatilai** [comment]
It's not an issue of format, it's a tag number issue and tag numbers are forever. The best chance is probably to let the whole tag die (replace by MIME types)

**Comment by pmatilai** [comment]
After some consideration, we'll consider this a request to enhance the Python bindings API to allow selectively dealing with extension vs raw header data like the C API can. Besides letting the tag die of old age, that's the only concrete thing we can do without breaking some usecase or the other.
